workflows/taxprofiler.nf

```diff
@@ -125,11 +125,49 @@ workflow TAXPROFILER {
         ch_shortreads_filtered = ch_shortreads_preprocessed
     }
 
+    /*
+        STEP: Run merging
+    */
+
+    if ( params.run_merging ) {
+
+        ch_reads_for_cat_branch = ch_shortreads_filtered
+            .mix( ch_longreads_preprocessed )
+            .map {
+                meta, reads ->
+                    def meta_new = meta.clone()
+                    meta_new.remove('run_accession')
+                    [ meta_new, reads ]
+            }
+            .groupTuple(by: 0)
+            .map {
+                meta, reads ->
+                    [ meta, reads.flatten() ]
+            }
+            .branch {
+                // we can't concatenate files if there is not a second run, we branch
+                // here to separate them out, and mix back in after for efficiency
+                cat: ( it[0]['single_end'] && it[1].size() > 1 ) || ( !it[0]['single_end'] && it[1].size() > 2 )
```

Comment:
Does `branch` actually allow for a closure? Would be great for understanding to split by `meta, reads` again.

Reply:
I never tried but I thought the code would be easier to read if it looked like this:

```groovy
.branch { meta, reads ->
    cat: (meta.single_end && reads.size() > 1) || (!meta.single_end && reads.size() > 2)
}
```

Reply:
Jesus, I was reading the curly brackets as round brackets... of course you can specify the variable names :facepalm: updating... I think I might be a tad overload atm. Thanks for the thorough review...

Should maybe do a code-quality review at a later stage to clean up things like that. I've used the `it` version of branch in  afew places.